src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,53 @@
+package mate.academy.service.impl;
+
+import java.util.Optional;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.AuthenticationService;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    private static final String CANT_LOGIN_EXCEPTION_MESSAGE =
+            "Can't login. Login or password is wrong.";
+    private static final String EXISTED_EMAIL_EXCEPTION_MESSAGE =
+            "Can't register user. This email is already used";
+    private static final String EMPTY_EMAIL_EXCEPTION_MESSAGE =
+            "Email can't be empty";
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> userFromDbOptional = userService.findByEmail(email);
+        if (userFromDbOptional.isEmpty()
+                || isPasswordCorrect(userFromDbOptional.get().getPassword(),
+                        HashUtil.hashPassword(password, userFromDbOptional.get().getSalt()))) {
+            throw new AuthenticationException(CANT_LOGIN_EXCEPTION_MESSAGE);
+        }
```

Comment:
Optional. We can separate this big check for two: 
1) check if the user not registered - inform user that no such user in database, 
2) if passwords are different - inform user that credentials are wrong.

It will simplify that code and it allow to make login process more clear for the user.